From 31584a1b6a62287ceb8abbc091c31164f99b87f7 Mon Sep 17 00:00:00 2001
From: James Turton <james.turton@gmx.com>
Date: Mon, 2 Mar 2026 22:28:38 +0100
Subject: [PATCH] Fix ac remmaping on R528-S3

Signed-off-by: James Turton <james.turton@gmx.com>
Upstream-Status: Pending [Not submitted to upstream yet]
---
 drivers/ram/sunxi/dram_sun20i_d1.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/ram/sunxi/dram_sun20i_d1.c b/drivers/ram/sunxi/dram_sun20i_d1.c
index 38379281d73..158c7083b8b 100644
--- a/drivers/ram/sunxi/dram_sun20i_d1.c
+++ b/drivers/ram/sunxi/dram_sun20i_d1.c
@@ -676,7 +676,8 @@ static void mctl_phy_ac_remapping(const dram_para_t *para,
 			case 8: cfg = ac_remapping_tables[2]; break;
 			case 9: cfg = ac_remapping_tables[3]; break;
 			case 10: cfg = ac_remapping_tables[5]; break;
-			case 11: cfg = ac_remapping_tables[4]; break;
+			case 11:
+			case 15: cfg = ac_remapping_tables[4]; break;
 			default:
 			case 12: cfg = ac_remapping_tables[1]; break;
 			case 13:
